### ISSUE TITLE
API EventOnHitUnit doesn't work, or I did it wrong?

### ISSUE THREAD

**Issue by xxx3456789** [LeagueSandbox/GameServer#807]
here is my Olaf Spell E code

```        
        private ISpell _spell;
        private void OnHitUnit(IAttackableUnit target, bool isCrit)
        {
            _spell.LowerCooldown(1);
        }       

        public void OnFinishCasting(IChampion owner, ISpell spell, IAttackableUnit target)
        {
            var adcost = owner.Stats.AttackDamage.Total;
            var healthcost = 7.5f + spell.Level * 13.5f + adcost * 0.3f;
            AddParticleTarget(owner, "olaf_recklessSwing_tar.troy", target, 1);
            AddParticleTarget(owner, "olaf_recklessSwing_tar_02.troy", target, 1);
            AddParticleTarget(owner, "olaf_recklessSwing_tar_03.troy", target, 1);
            AddParticleTarget(owner, "olaf_recklessSwing_tar_04.troy", target, 1);
            AddParticleTarget(owner, "olaf_recklessSwing_tar_05.troy", target, 1);
            var ad = owner.Stats.AttackDamage.Total * 0.4f;
            var damage = 25f + spell.Level * 45 + ad;

            target.TakeDamage(owner, damage, DamageType.DAMAGE_TYPE_TRUE, DamageSource.DAMAGE_SOURCE_SPELL, false);

            ApiEventManager.OnHitUnit.AddListener(this, owner, OnHitUnit);

            if (target.IsDead)
            {
                owner.Stats.CurrentHealth += healthcost;
            }
        }
``` 
I don't know it is my problem or just API doesn't work?

**Comment by cajthamlmatej** [comment]
You can see [here](https://github.com/LeagueSandbox/GameServer/blob/aeb35b5d27db66e833097a8217f939405734c092/GameServerLib/API/ApiEventManager.cs#L185) that is not done yet - event is not implemented anywhere.

**Comment by xxx3456789** [comment]
so it unable to use yet, right?

**Comment by wildbook** [comment]
Right. It's planned but not implemented.

**Comment by danil179** [comment]
~~Don't open issue about problems with your code. Issues are for existing problems in the repo, what you could do is to discuss in LS dev channels about your code and when you think your code is ready, open PR to merge it.~~

This is valid issue.

**Comment by wildbook** [comment]
IMO this would be a valid issue as it's something that (unless you look at the implementation) looks implemented but isn't. If nothing else it's something to reference in whichever PR implements it I guess.

I can understand your side of it as well though, guess it depends on how we want to treat issues.

**Comment by lzardy** [comment]
Resolved as of #1057.
